### PR TITLE
Fix for #1973 - full tuner broken for heterogeneous GPUs and auto precision detection

### PR DESCRIPTION
**Pull request by ihavnoid**
See #1973 

--full-tuner implies --tune-only
--full-tuner requires an explicit precision

### CODE REVIEW

**Comment by Hersmunch** [comment]
Are the changes to tfprocess meant to be included?

**Comment by ihavnoid** [comment]
Argh, my bad.  Will push a change that reverts my experiments.

**Comment by ihavnoid** [comment]
@Hersmunch - okay reverted the tensorflow code changes.

File: src/Leela.cpp [hunk 1]
```diff
@@ -238,6 +243,15 @@ static void parse_commandline(int argc, char *argv[]) {
             exit(EXIT_FAILURE);
         }
     }
+    if (cfg_precision == precision_t::AUTO) {
+        // Auto precision is not supported for tune only cases.
+        // Since full tuner implies tune only, check that too.  #1973
+        if (cfg_sgemm_exhaustive || cfg_tune_only) {
```
Comment (Hersmunch) [inline src/Leela.cpp]: Is there a good reason for repeating the full tuner implying tune only handling? I’d prefer if the decision is made only in one place.

Comment (ihavnoid) [reply src/Leela.cpp]: Yes I think it makes sense.  I'll push achange.

File: src/Leela.cpp [hunk 2]
```diff
@@ -238,6 +243,15 @@ static void parse_commandline(int argc, char *argv[]) {
             exit(EXIT_FAILURE);
         }
     }
+    if (cfg_precision == precision_t::AUTO) {
+        // Auto precision is not supported for tune only cases.
+        // Since full tuner implies tune only, check that too.  #1973
```
Comment (Hersmunch) [inline src/Leela.cpp]: You missed the comment :)

**Comment by gcp** [comment]
Did some minor edits.

Remove references to issue numbers - only makes sense if you can look it up on github, and you might as well use git blame to understand why code is there.

Replaced std::list by std::vector.

Pushed as 4a044d9.

But I will have to back this out: it's not compatible with AutoGTP, which needs to tune once but has no idea about what precision it should use.

**Pull request merged**